scripts/drift_corr_ion.py

```diff
@@ -0,0 +1,163 @@
+    parser.add_argument("--roi", dest="roi", required=True,
+                        help="e-beam ROI positions (ltrb, relative to the SEM "
+                             "field of view)")
+    parser.add_argument("--dwell", dest="dwell_time", type=float,
+                        default=10e-06, help="(s) FIB stays for this much duration while scanning")
+    parser.add_argument("--nb", dest="nb_images", type=int,
+                        default=10, help="total images acquired for drift correction")
+    parser.add_argument("--delay", dest="delay_time", type=int,
+                        default=5, help="(s) time duration between the acquisition of "
+                                        "consecutive images")
+    parser.add_argument(dest="base",
+                        help="filename of the image to be stored")
+
+    options = parser.parse_args(args[1:])
+
+    try:
+        escan = None
+        detector = None
+
+        # find components by their role
+        for c in model.getComponents():
+            if c.role == "e-beam":
+                escan = c
+            elif c.role == "se-detector":
+                detector = c
+
+        if not all([escan, detector]):
+            logging.error("Failed to find all the components")
+            raise KeyError("Not all components found")
+    except:
+        logging.exception("Unexpected error while performing action.")
+        return 127
+
+    roi = conversion.reproduce_typed_value([1.0], options.roi)
+    if not all(0 <= r <= 1 for r in roi):
+        raise ValueError("roi values must be between 0 and 1")
+
+    escan.blanker.value = None
+    p = Path(options.base)
+    nb_images = options.nb_images
+    delay_time = options.delay_time
+    dwell_time = options.dwell_time  # s
+    drift_est = drift.AnchoredEstimator(escan, detector,
+                                        roi,
+                                        dwell_time, max_pixels=512 ** 2, follow_drift=False)
+    logging.info('Dwell time(s): %s', dwell_time)
+    logging.info('roi: %s', roi)
+    logging.info('Hfov(m): %s', escan.horizontalFoV.value)
+    logging.info('current(A): %s', escan.current.value)
+    logging.info('MAX res(px): %s', math.sqrt(MAX_PIXELS))
+    logging.info('Acquire time(s): %s', delay_time)
+    logging.info('file name by: %s', time.strftime("%m%d-%H%M"))
+    logging.info('Max images: %s', nb_images)
+    logging.info('Pixel size(m): %s', escan.pixelSize.value)
+
+    i = 0
+    row_drift = []
+    col_drift = []
+
+    for i in range(nb_images):
+        logging.debug("IMAGE %s", i)
+
+        # Acquire an image at the given location (RoI)
+        drift_est.acquire()
+        da = drift_est.raw[-1]
+
+        # Save the Acquired Image
+        # (For debugging purposes, have an option to store the image, with current date and time in the filename)
+        timestr = time.strftime("%m%d-%H%M") + "_" + str(i)
+        filename = "{0}{2}{1}".format(p.stem, p.suffix, timestr)
```

Comment:
It'd be easier to read if you ordered it 0, 1, 2. If you use the f-string, it'll be even easier to read.